sdks/java/io/rabbitmq/src/main/java/org/apache/beam/sdk/io/rabbitmq/RabbitMqIO.java

```diff
@@ -462,12 +449,14 @@ public boolean start() throws IOException {
     @Override
     public boolean advance() throws IOException {
       try {
-        QueueingConsumer.Delivery delivery = consumer.nextDelivery(1000);
+        Channel channel = connectionHandler.getChannel();
+        // we consume message without autoAck (we want to do the ack ourselves)
+        GetResponse delivery = channel.basicGet(queueName, false);
```

Comment:
Important change: previously the caller would wait for up to 1s for a message to be available. This PR's implementation would return `false` immediately (technically after a network hop). The semantics of this change is unclear to me but from what I can tell, this should be "fine" at worst and an improvement at best.

Reply:
@chamikaramj @dpmills @reuvenlax might have good guidance here